### PR TITLE
Fix await completion provider to handle non-"true" local function declarations

### LINKED ISSUE #56006: `Make container 'async'` try to insert an `async` at strange place
**Version Used**: 
VS2022 17.0.0 Preview 3.1

**Steps to Reproduce**:

GIF1

**Expected Behavior**:
It should make the parent method async.

**Actual Behavior**:
It just insert an `async` before `await`.

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #56006

### CODE REVIEW

File: src/Features/CSharp/Portable/Completion/CompletionProviders/AwaitCompletionProvider.cs [hunk 1]
```diff
@@ -50,6 +48,23 @@ private protected override int GetSpanStart(SyntaxNode declaration)
...
+            // }
+            // For this case, we're interested in putting async in front of Method()
+            if (node is not LocalFunctionStatementSyntax localFunction ||
+                localFunction.ExpressionBody is { ArrowToken.RawKind: not 0 } ||
+                localFunction.Body is { OpenBraceToken.RawKind: not 0 })
+            {
+                return node;
+            }
+
+            return node.Parent?.FirstAncestorOrSelf<SyntaxNode>(node => node.IsAsyncSupportingFunctionSyntax());
```
Comment (davidwengier) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/AwaitCompletionProvider.cs:67]: This is hard to parse, because the comment talks about the exception case, but that case is actually handled by the return at the end of the method. Couldn't this be re-written as:

```suggestion
            if (node is LocalFunctionStatementSyntax { ExpressionBody: null, Body: null })
            {
                return node.Parent?.FirstAncestorOrSelf<SyntaxNode>(node => node.IsAsyncSupportingFunctionSyntax());
            }

            return node;
```

Comment (Youssef1313) [reply src/Features/CSharp/Portable/Completion/CompletionProviders/AwaitCompletionProvider.cs:67]: I think it *should* work, and it's much much better!

**Review by davidwengier — APPROVED** [review]

**Comment by davidwengier** [comment]
Thanks @Youssef1313!

At least someone is writing code on this US long weekend 😛

**Comment by MaStr11** [comment]
This bug was already found as part of #55817. It was fixed in 5ff9d09edd4f685b153ee39c4deb15bbdb1faddf. I merged main into that PR and removed the changes made here, but kept the test.
